packages/engine/src/interaction/systems/InteractiveSystem.ts

```diff
@@ -45,14 +48,27 @@ export default async function InteractiveSystem(world: World): Promise<System> {
 
   return () => {
     for (const entity of interactiveQuery.enter(world)) {
-      if (!hasComponent(entity, BoundingBoxComponent) && hasComponent(entity, Object3DComponent)) {
+      if (!hasComponent(entity, BoundingBoxComponent)) {
         createBoxComponent(entity)
       }
       if (!getInteractUI(entity)) {
         createInteractUI(entity)
       }
     }
 
+    for (const entity of interactiveQuery.exit(world)) {
+      if (hasComponent(entity, BoundingBoxComponent)) {
+        // TODO: Does the Box3 object need to be destroyed before this?
+        removeComponent(entity, BoundingBoxComponent)
+      }
+      if (hasComponent(entity, InteractiveFocusedComponent)) {
+        removeComponent(entity, InteractiveFocusedComponent)
+      }
+      if (hasComponent(entity, SubFocusedComponent)) {
+        removeComponent(entity, SubFocusedComponent)
+      }
```

Comment:
are these `hasComponent` checks necessary if it's removing the component anyways? if it's needed, perhaps we should add the `hadComponent` check to the `removeComponent` function instead

Reply:
@NateTheGreatt Does bitecs allow `removeComponent` if the entity does not have the component?

Reply:
Yeah I was of the assumption that bitECS does not allow removeComponent if the component does not exist. But I just ran a test it does not throw any errors. So I will update the code and remove the extra hasComponent checks.

Reply:
Done